front/app/containers/IdeasNewPage/index.tsx

```diff
@@ -251,11 +253,52 @@ class IdeasNewPage extends PureComponent<Props & WithRouterProps, State> {
           });
         }
       } catch (error) {
+        if (process.env.NODE_ENV === 'development') console.log(error);
+        const apiErrors = get(error, 'json.errors');
+        const profanityApiError = apiErrors.base.find(
+          (apiError) => apiError.error === 'includes_banned_words'
+        );
+
+        if (profanityApiError) {
+          const titleProfanityError = profanityApiError.blocked_words.some(
+            (blockedWord) => blockedWord.attribute === 'title_multiloc'
+          );
+          const descriptionProfanityError = profanityApiError.blocked_words.some(
+            (blockedWord) => blockedWord.attribute === 'body_multiloc'
+          );
+
+          if (titleProfanityError) {
+            this.globalState.set({
+              titleProfanityError,
+            });
+          }
+
+          if (descriptionProfanityError) {
+            this.globalState.set({
+              descriptionProfanityError,
+            });
+          }
+        }
+
         this.globalState.set({ processing: false, submitError: true });
       }
     }
   };
 
+  onTitleChange = (title: string) => {
+    this.globalState.set({
+      title,
```

Comment:
I had to add updating of title in globalState here, despite it already getting updated whenever you hit submit. Reason for that: if I remove the profanityError on change here, it'll send new props (because of titleProfanityError) to NewIdeaForm/ and IdeaForm. Because globalState wasn't updated yet with the title, it would send the old title to IdeaForm, which clashes with the state hold in IdeaForm.

Going over IdeasNewPage > NewIdeaForm > IdeaForm is was wondering why we have globalState + a separate local state for each component. We are basically having 2 separate states for each component. Why don't we hold state in the component that handles the API calls? That would be 1 state to manage vs. 4 states now.

Reply:
@DavidCnoops thoughts here?